### PR TITLE
Refs #10690 - only create operating system for repo that was synced

### PR DESCRIPTION
**Pull request by jlsherrill**
rather than for all repos in the content view/env

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test]

File: app/lib/actions/katello/foreman/content_update.rb [hunk 1]
```diff
@@ -17,15 +18,18 @@ def plan(environment, content_view)
...
           environment  = ::Katello::KTEnvironment.find(input[:environment_id])
           content_view = ::Katello::ContentView.find(input[:content_view_id])
-          ::Katello::Foreman.update_puppet_environment(content_view, environment)
+          repository = ::Katello::Repository.find(input[:repository_id]) if input[:repository_id]
 
-          content_view.version(environment).repos(environment).each do |repo|
-            if distribution = repo.bootable_distribution
+          if content_view.default?
+            distribution = repository.try(:bootable_distribution)
+            if distribution
```
Comment (ehelms) [inline app/lib/actions/katello/foreman/content_update.rb]: Could combine the two statements:

`if (distribution = repository.try(:bootable_distribution)`

File: app/lib/actions/katello/foreman/content_update.rb [hunk 2]
```diff
@@ -17,15 +18,18 @@ def plan(environment, content_view)
         def finalize
           environment  = ::Katello::KTEnvironment.find(input[:environment_id])
           content_view = ::Katello::ContentView.find(input[:content_view_id])
-          ::Katello::Foreman.update_puppet_environment(content_view, environment)
+          repository = ::Katello::Repository.find(input[:repository_id]) if input[:repository_id]
 
-          content_view.version(environment).repos(environment).each do |repo|
-            if distribution = repo.bootable_distribution
+          if content_view.default?
```
Comment (ehelms) [inline app/lib/actions/katello/foreman/content_update.rb]: Might as well check for repository here as well? `if repository && content_view.default?` or do you not want to want to process that else statement if there is no repository?

Comment (jlsherrill) [inline app/lib/actions/katello/foreman/content_update.rb]: correct, i do not

**Comment by ehelms** [comment]
Out of curiosity, I see that it was doing a ` Redhat.find_or_create_operating_system(distribution)` before, so if when the repo was synced the operating system was initially created, why was it attempting to recreate it for clones?

**Comment by jlsherrill** [comment]
I don't know of any reason why it was trying to create it.  We don't create install media for non-Library repos, so even if it had been deleted it wouldn't need to be recreated for anything

**Comment by ehelms** [comment]
Seems like an unsolved mystery, but this code makes sense so ACK

**Pull request merged**